### PR TITLE
Fix displacement of colorbar for eps with bbox_inches='tight'

### LINKED ISSUE #25176: [Bug]: Colorbar is displaced when saving as .eps with bbox_inches='tight'
### Bug summary
I am plotting with `imshow()` and create a colorbar. When saving the figure as .eps with `savefig()` and option `bbox_inches='tight'`, the colorbar (without the spines) is displaced.

### Code for reproduction
```python
import matplotlib.pyplot as plt

fig = plt.figure()
plt.imshow([[1, 1], [1, 1]])
plt.colorbar()
plt.savefig("test.eps", bbox_inches='tight')
plt.show()
```

### Expected outcome
Colorbar should be placed inside the spines as when using `plt.show()`. Also, saving as .png instead of .eps works correctly.

### Actual outcome
test

### Additional information
The bug does not happen when using Matplotlib version 5.2.3. 

### Operating system
Windows 10

### Matplotlib version
3.6.3

### Matplotlib backend
module://backend_interagg

### Python version
3.11.0

### PR DESCRIPTION
**Pull request by ksunden**
Co-authored-by: Antony Lee <anntzer.lee@gmail.com>

Closes #25176

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
I'm going to say that between this being @anntzer 's patch, @ksunden writing it up and adding a test and my review we have crossed the "2 reviewers" threshold and will merge this when green.

**Review by anntzer — APPROVED** [review]
Post ci.

**Review by jklymak — APPROVED** [review]

File: lib/matplotlib/tests/test_backend_ps.py [hunk 1]
```diff
@@ -327,3 +328,11 @@ def test_path_collection():
                                      facecolors='yellow', offsets=offsets)
     ax.add_collection(pc)
     ax.set_xlim(0, 1)
+
+
+@image_comparison(["colorbar_shift.eps"], savefig_kwarg={"bbox_inches": "tight"})
```
Comment (oscargus) [inline lib/matplotlib/tests/test_backend_ps.py:333]:
```suggestion
@image_comparison(["colorbar_shift.eps"], savefig_kwarg={"bbox_inches": "tight"}, style='mpl20')
```

Comment (ksunden) [reply lib/matplotlib/tests/test_backend_ps.py]: Updated

**Pull request merged**